components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -56,8 +57,13 @@ impl<S: Storage> BatchIndexScanExecutor<S> {
         // Note 3: Currently TiDB may send multiple PK handles to TiKV (but only the last one is
         // real). We accept this kind of request for compatibility considerations, but will be
         // forbidden soon.
-
-        let is_int_handle = columns_info.last().map_or(false, |ci| ci.get_pk_handle());
+        let decode_partition_id = columns_info
+            .last()
+            .map_or(false, |ci| ci.get_column_id() == table::EXTRA_PID_COL_ID);
```

Comment:
Please give a doc link or code comment of this behavior, it is a new layout of pushed-down column infos.